packages/wrangler/src/user.tsx

```diff
@@ -245,6 +245,15 @@ const getCloudflareAccountIdFromEnv = getEnvironmentVariableFactory({
   deprecatedName: "CF_ACCOUNT_ID",
 });
 
+/**
+ * Try to read the Wrangler oauth callback URL from the environment.
+ */
+ const getWranglerOauthCallbackUrl = getEnvironmentVariableFactory({
+  variableName: "WRANGLER_OAUTH_CALLBACK_URL",
+  defaultValue: "http://localhost:8976/oauth/callback"
```

Comment:
I think you'll only want to replace the root part (`http://localhost:8976`), not the whole url, no?

Reply:
I was thinking about cases when the platform has `/oauth/callback` restricted or reserved for some reason, when wrangler is running behind a proxy.

Reply:
Is that a situation that actually happens? Because they have to hit our server endpoint. If we're letting them specify a full URL, then we also have to change where we read the data, so that'll have to be part of this PR too. 

Glad we have prereleases to test this.

Reply:
That is a good point, gonna change it to support the root part only for now.

Reply:
I don't follow that reasoning @threepointone. 
This URL is the one that is exposed by Wrangler, not the one exposed by our OAuth API.
Why would the latter need to change?

Reply:
@petebacondarwin it would also need to be changed, at least, here https://github.com/cloudflare/wrangler2/blob/0a79d75e6aba11a3f0d5a7490f1b75c9f3e80ea8/packages/wrangler/src/user.tsx#L938

Reply:
Ah! I see. Fair enough.